### ISSUE TITLE
Have location.occupant and piece.location watch each other

### ISSUE THREAD

**Issue by TAGraves** [TAGraves/ES6-Chess#1]
As it is now, all of the functions that change one of these also change the other. But why not use a setter to make them watch each other?

``` javascript
class Location {
  set occupant (piece) {
    this.occupant = piece;
    piece.location = this;
  }
}
```

``` javascript
class Piece {
  set location (location) {
    this.location = location;
    location.occupant = this;
  }
}
```

This may fail testing at the moment because of the en passant method which does some tricky stuff.

**Comment by TAGraves** [comment]
Solution:

Make a setter for Piece().isCaptured that sends it to Board.out and use that in the en passant code.

**Comment by TAGraves** [comment]
Also, use `_occupant` and so on to prevent an infinite loop of setters.

**Comment by TAGraves** [comment]
Fixed in c333c8aae202c51e83242d6066bb494ab4f4135e
